pywbem/cim_operations.py

```diff
@@ -453,11 +459,23 @@ class WBEMConnection(object):
         :class:`~pywbem.BaseOperationRecorder`, each operation
         that is executed on this connection will be recorded by
         invoking its :meth:`~pywbem.BaseOperationRecorder.record` method.
+
+      use_pull_operations (:class:`py:bool`):
+        Attribute that defines whether the WBEM server defined by this
+        connection has or is to use the pull operations
+
+        if `None` no pull operation has been attempted on this connection
+        If `True` the user has specified pull operations only or the client
+        has received a successful response to a Iter... method.
+        If `False` the  the user set use_pull_operations=False in the WBEM
+        connection creation or the server has responded with not supported
+        indicating that the pull operations are not supported on the server.
+
```

Comment:
I can meanwhile understand why you had named the instance variable `has_pull_operations` at some point (if i understood you correctly in the call): After the first call to an `Iter...()` method it is set to reflect the capability of the server, but only if it was not initially passed as `False`. I think naming the instance variable `use_pull_operations` like it is currently is better, because it does not truly reflect the server capability, but what is used in the `Iter...()` methods. No change needed, I just wanted to provide my thoughts on this.

Comments on the description:

1. On "... the WBEM server ... has or is to use ...": It is the client that uses them, not the server.

2. The description of the three values mostly repeats the algorithm that is implemented. I think it should be written by focusing on the use of pull.

3. As an instance variable, I prefer wording that explains what the variable indicates instead of what it controls or defines.

How about this description:

```
        Indicates whether the client will attempt the use of pull operations in any `Iter...()` methods.

        This variable is initially set from the same-named input argument, and is
        adjusted when `Iter...()` methods are invoked, dependent on the support
        for pull operations in the WBEM server.
        After adjusting the variable, it is either `True` or `False` and indicates whether
        pull operations will be used in any `Iter...()` methods.
```